service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -198,4 +216,140 @@ private void run2177(CaseDetails caseDetails) {
+
+    private void runAM(CaseDetails caseDetails) {
+        var migrationId = "DFPL-AM";
+
+        CaseData caseData = getCaseData(caseDetails);
+
+        // 1. cleanup from past migration
+        judicialService.deleteAllRolesOnCase(caseData.getId());
+
+        // 2. Add UUIDs for judges + legal advisers
+        Judge allocatedJudge = caseData.getAllocatedJudge();
+        if (!isEmpty(allocatedJudge) && !isEmpty(allocatedJudge.getJudgeEmailAddress())) {
+            String email = allocatedJudge.getJudgeEmailAddress();
+            Optional<String> uuid = judicialService.getJudgeUserIdFromEmail(email);
+            // add the UUID to the allocated judge and save on the case
+
+            if (uuid.isEmpty()) {
+                log.info("Could not find judge UUID, caseId={}, allocatedJudge, ejudiciary={}, judgeTitle={}",
+                    caseData.getId(), email.toLowerCase().endsWith("@ejudiciary.net"), allocatedJudge.getJudgeTitle());
+            }
+
+            uuid.ifPresent(s -> caseDetails.getData().put("allocatedJudge", allocatedJudge.toBuilder()
+                .judgeJudicialUser(JudicialUser.builder()
+                    .idamId(s)
+                    .build())
+                .build()));
+        }
+
+        List<Element<HearingBooking>> hearings = caseData.getAllNonCancelledHearings();
+        List<Element<HearingBooking>> modified = hearings.stream()
+            .map(el -> {
+                HearingBooking val = el.getValue();
+                if (!isEmpty(val.getJudgeAndLegalAdvisor())
+                    && !isEmpty(val.getJudgeAndLegalAdvisor().getJudgeEmailAddress())) {
+
+                    String email = val.getJudgeAndLegalAdvisor().getJudgeEmailAddress();
+                    Optional<String> uuid = judicialService.getJudgeUserIdFromEmail(email);
+                    if (uuid.isPresent()) {
+                        el.setValue(val.toBuilder()
+                            .judgeAndLegalAdvisor(val.getJudgeAndLegalAdvisor().toBuilder()
+                                .judgeJudicialUser(JudicialUser.builder()
+                                    .idamId(uuid.get())
+                                    .build())
+                                .build())
+                            .build());
+                        return el;
+                    } else {
+                        log.info("Could not find judge UUID, caseId={}, hearingId={}, ejudiciary={}, judgeTitle={}",
+                            caseData.getId(), el.getId(), email.toLowerCase().endsWith("@ejudiciary.net"),
+                            val.getJudgeAndLegalAdvisor().getJudgeTitle());
+                    }
+                }
+                return el;
+            }).toList();
+
+        if (!hearings.isEmpty()) {
+            // don't add an empty array if there weren't any hearings beforehand
+            caseDetails.getData().put("hearingDetails", modified);
+        }
+        caseDetails.getData().put("hasBeenAMMigrated", "Yes");
```

Comment:
I guess the existence of  `hasBeenAMMigrated` will be checked in migration tool. Shall we check it in `runAM` as well?

Reply:
it must be present or the elasticsearch query for the rollback won't select the case, doesn't matter if it's present or not for the migration, as it'll delete + recreate roles anyways (and some cases will still have it from before, some wont and they all need migrating this time around!)